### PR TITLE
Scaled bipedal walker reward

### PR DESCRIPTION
**Pull request by matteobettini**

### CODE REVIEW

File: popgym/envs/bipedal_walker.py [hunk 1]
```diff
@@ -561,18 +572,38 @@ def step(self, action: np.ndarray):
...
+        if self.steps == self.max_episode_length:
+            done = True
+
         return np.array(state, dtype=np.float32), reward, done, {}
 
+    def reward_transform(self, reward):
+        low, high = (
+            FAIL_REWARD
+            - self.action_space.shape[0] * TORQUE_REWARD_MULTIPLIER * MOTORS_TORQUE,
+            MAX_DELTA_POS * 130 / SCALE,
```
Comment (smorad) [inline popgym/envs/bipedal_walker.py:600]: Where does the 130 come from?

File: popgym/envs/bipedal_walker.py [hunk 2]
```diff
@@ -61,6 +61,12 @@
 TERRAIN_STARTPAD = 20  # in steps
 FRICTION = 2.5
 
+FAIL_REWARD = -100
+TORQUE_REWARD_MULTIPLIER = 0.00035
+MAX_DELTA_POS = (
```
Comment (smorad) [inline popgym/envs/bipedal_walker.py:66]: You should set this in init so it is a function of `max_episode_length`. Currently, if the length changes then the reward no longer sums to one.

**Review by smorad — CHANGES_REQUESTED** [review]
Looks great, just a few small comments.

File: popgym/envs/bipedal_walker.py (hunk 2 above)
Comment (matteobettini) [reply popgym/envs/bipedal_walker.py:66]: The max delta pos obtainable by a worker is independent of the `max_episode_length`. I calculate it using the data that was in the original environment (all terrain covered in 1600 steps). Are you sure the reward doesn't sum to 1? i tested it with diefferen legths (100,200,300) and the heuristic

File: popgym/envs/bipedal_walker.py (hunk 1 above)
Comment (matteobettini) [reply popgym/envs/bipedal_walker.py:600]: It comes from the reward function. It is yet another constant that they did not define, i ll define it

**Comment by smorad** [comment]
Awesome, thanks!

**Comment by smorad** [comment]
Awesome, thanks!

**Pull request merged**